From 1f88e2b09a6a1f34d9a695d7df6521cab2576c7d Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 6 May 2016 01:10:58 -0400
Subject: [PATCH 20/22] disable non-implemented functions

It is minimal installation for now, the non-implemented
functions will be disabled.

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/ui/gui/hubs/__init__.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/ui/gui/hubs/__init__.py b/pyanaconda/ui/gui/hubs/__init__.py
index 2563300..d4fe36b 100644
--- a/pyanaconda/ui/gui/hubs/__init__.py
+++ b/pyanaconda/ui/gui/hubs/__init__.py
@@ -202,7 +202,14 @@ class Hub(GUIObject, common.Hub):
         self._updateContinue()
 
     def _updateCompleteness(self, spoke, update_continue=True):
-        spoke.selector.set_sensitive(spoke.sensitive and spoke.ready)
+        log.info("_updateCompleteness %s" % (spoke.mainWidgetName))
+        # It is minimal installation for now, the non-implemented
+        # functions will be disabled.
+        if spoke.mainWidgetName not in ["storageWindow", "passwordWindow"]:
+            spoke.selector.set_sensitive(False)
+        else:
+            spoke.selector.set_sensitive(spoke.sensitive and spoke.ready)
+
         spoke.selector.set_property("status", spoke.status)
         spoke.selector.set_tooltip_markup(escape_markup(spoke.status))
         spoke.selector.set_incomplete(not spoke.completed and spoke.mandatory)
-- 
2.8.1

